src/Tools/DynamoShapeManager/Utilities.cs

```diff
@@ -410,7 +410,26 @@ public static string GetGeometryFactoryPath2(string rootFolder, Version version)
 
             return assemblyPath;
         }
-
+        /// <summary>
+        /// This method will return the path to the GeometryFactory assembly location for a requested version of the geometry library.
+        /// This method is tolerant to the requested version in that it will attempt to locate an exact or lower version of the GeometryFactory assembly.
+        /// </summary>
+        /// <param name="rootFolder">Full path of the directory that contains 
+        /// LibG_xxx_y_z folder, where 'xxx y z' represents the library version of asm. In a 
+        /// typical setup this would be the same directory that contains Dynamo 
+        /// core modules. This must represent a valid directory - it cannot be null.</param>
+        /// <param name="tolerantVersion">Version number of the targeted geometry library.
+        /// If the resulting assembly does not exist, this method will look for a lower version match.
+        /// This parameter cannot be null. </param>
+        /// <returns>The full path to GeometryFactoryAssembly assembly.</returns>
+        /// 
+        public static string GetGeometryFactoryPathTolerant(string rootFolder, Version tolerantVersion){
+            if(tolerantVersion != null && rootFolder != null)
+            {
+                return Path.Combine(Utilities.GetLibGPreloaderLocation(tolerantVersion, rootFolder), Utilities.GeometryFactoryAssembly);
```

Comment:
The code itself looks fine but I was expecting us to merge some code between `GetGeometryFactoryPath2` and `GetGeometryFactoryPathTolerant` because in the end either DynamoCore or our client would only be available to call one of them. But for now the code path between these two are quite different except for the fall back logic? Do you think if we also still need those folder check logic in this new function?

Reply:
1. I find these two methods both potentially useful - they do different things, and I think it's always useful to have an explicit method to avoid unknown behavior which is why I did not obsolete the other method.

2. Sandbox AFAIK does not call these methods at all.

3. The end result to the client is like you said that it appears it does a fallback - it DOES have different edge case behavior though - it returns a empty string instead of exceptions when it encounters bad input - that's kind of what I was asking about in the test cases? I'm okay with the more slow failure because this is a `tolerant` version of the method, but open to other opinions for sure.

Reply:
Sure, I don't mean to obsolete the other one. I thought we want to include the folder check logic in the new API as well because of some legacy bug, if you think it is OK and good enough for the clients then this version should be fine to use. Do you plan to make any change in client code as an example usage of this API later when nuget is up?

Reply:
I actually don't see any reason to keep `GetGeometryFactoryPath2` after the addition of this new API and I think we should add the folder check exceptions to the new API. I don't see reasons to remove them from here. `GetGeometryFactoryPath2` can give unexpected results (in combination with other API's) like we've seen in the Revit case so I think it should be obsoleted.